lib3h/src/p2p/p2p_trait.rs

```diff
@@ -0,0 +1,21 @@
+use crate::p2p::p2p_event::P2pEvent;
+use holochain_lib3h_protocol::{DidWork, Lib3hResult};
+
+/// Composition of a Connection, Dht, RuntimeStore, PeerStore, and DataStore
+/// to provide a higher-level Connection interface.
+pub trait P2p {
+    /// Lifecycle
+    fn connect(&mut self, url: String) -> Lib3hResult<()>;
+    fn close(&self, peer_address: String) -> Lib3hResult<()>;
+    /// Comms
+    fn publish_reliable(&self, peer_list: Vec<String>, data: Vec<u8>) -> Lib3hResult<()>;
```

Comment:
Function parameters should accept references, and probably need to be `&mut self`:

```rust
fn publish_reliable(&mut self, peer_list: &[&Address], data: &[u8]) -> Lib3hResult<()>;
```

Reply:
Actually... if `Address` maps to `String` we probably need an `AddressRef` that maps to `str`:

```rust
pub type AddressRef = str;
fn publish_reliable(&mut self, peer_list: &[&AddressRef], data: &[u8]) -> Lib3hResult<()>;
```